chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -4953,6 +4955,60 @@ func TestCCIPRouter(t *testing.T) {
 						testcase.RunStateValidations(t)
 					})
 				}
+
+				for _, testcase := range priceUpdatesCases {
+					if testcase.skipWithNoMerkleRoot {
+						continue
+					}
+					t.Run("prices only: "+testcase.Name, func(t *testing.T) {
+						report := ccip_offramp.CommitInput{
+							MerkleRoot:   nil,
+							PriceUpdates: testcase.PriceUpdates,
+						}
+
+						var reportContext [2][32]byte
+						if testcase.ReportContext != nil {
+							reportContext = *testcase.ReportContext
+						} else {
+							reportContext = ccip.NextCommitReportContext()
+						}
+
+						sigs, err := ccip.SignCommitReport(reportContext, report, signers)
+						require.NoError(t, err)
+
+						transmitter := getTransmitter()
+
+						raw := ccip_offramp.NewCommitPriceOnlyInstruction(
+							reportContext,
+							testutils.MustMarshalBorsh(t, report),
+							sigs.Rs,
+							sigs.Ss,
+							sigs.RawVs,
+							config.OfframpConfigPDA,
+							config.OfframpReferenceAddressesPDA,
+							transmitter.PublicKey(),
+							solana.SystemProgramID,
+							solana.SysVarInstructionsPubkey,
+							config.OfframpBillingSignerPDA,
+							config.FeeQuoterProgram,
+							config.FqAllowedPriceUpdaterOfframpPDA,
+							config.FqConfigPDA,
+						)
+
+						for _, pubkey := range testcase.RemainingAccounts {
+							raw.AccountMetaSlice.Append(solana.Meta(pubkey).WRITE())
+						}
+
+						instruction, err := raw.ValidateAndBuild()
+						require.NoError(t, err)
+
+						tx := testutils.SendAndConfirmWithLookupTables(ctx, t, solanaGoClient, []solana.Instruction{instruction}, transmitter, rpc.CommitmentConfirmed, offrampLookupTable, common.AddComputeUnitLimit(MaxCU))
+						commitEvent := ccip.EventPriceOnlyCommitReportAccepted{}
+						require.NoError(t, common.ParseEvent(tx.Meta.LogMessages, "PriceOnlyCommitReportAccepted", &commitEvent, config.PrintEvents))
+
+						require.Equal(t, commitEvent.PriceUpdates, testcase.PriceUpdates)
+					})
```

Comment:
This seems to only check the offramp part, could you please also validate that the prices are correctly propagated to FeeQuoter? E.g. reading fee quoter PDAs is fine, and you may do so with `testcase.RunStateValidations(t)` and just run the same validations of the cases where there is an accompanying root.